File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/boot_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class BootCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(
            BootCC.conditions(), condition,
        )

    @staticmethod
    def conditions() -> Set[Condition]:
        return {
            Condition.Z,
            Condition.NZ,
            Condition.XZ,
            Condition.XNZ,
            Condition.SZ,
            Condition.SNZ,
            Condition.SPL,
            Condition.SMI,
            Condition.TRUE,
            Condition.FALSE,
        }

File: golang/uPIMulator/src/linker/kernel/instruction/cc/boot_cc.go
package cc

import (
	"errors"
)

type BootCc struct {
	condition Condition
}

func (this *BootCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		Z:     true,
		NZ:    true,
		XZ:    true,
		XNZ:   true,
		SZ:    true,
		SNZ:   true,
		SPL:   true,
		SMI:   true,
		TRUE:  true,
		FALSE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *BootCc) Condition() Condition {
	return this.condition
}
